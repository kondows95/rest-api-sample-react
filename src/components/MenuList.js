import React from 'react'
import PropTypes from 'prop-types'
import { makeStyles } from '@material-ui/core/styles'
import { List, ListItem, ListItemText, ListItemIcon, Divider} from '@material-ui/core'
import {
    Photo as PhotoIcon, 
    Brush as BrushIcon, 
    Face as FaceIcon,
    ShoppingCart as ShoppingCartIcon,
    Accessibility as AccessibilityIcon,
    Payment as PaymentIcon
} from '@material-ui/icons'
import { Link } from 'react-router-dom'

const useStyles = makeStyles(theme => ({
  link: {
    textDecoration: "none",
    color: theme.palette.primary.main
  },
}))


const CategoryList = ({ categories, setCategoryId, handleDrawerClose }) => {
  const classes = useStyles()
  return (
    <React.Fragment>
      <List>
        <Link to = '/categories' className={classes.link}>
          <ListItem button className="icon">
            <ListItemIcon ><BrushIcon /></ListItemIcon>
            <ListItemText>Categories</ListItemText>
          </ListItem>
        </Link>
        <Link to = '/items' className={classes.link}>
          <ListItem button className="icon">
            <ListItemIcon><PhotoIcon /></ListItemIcon>
            <ListItemText>Items</ListItemText>
          </ListItem>
        </Link>
      </List>
      
      
      <Divider />
      <List>
        <Link to = '/dummycheckout' className={classes.link}>
          <ListItem button className="icon">
            <ListItemIcon><PaymentIcon /></ListItemIcon>
            <ListItemText>Checkout</ListItemText>
          </ListItem>
        </Link>
        <Link to = '/orders' className={classes.link}>
          <ListItem button className="icon">
            <ListItemIcon><ShoppingCartIcon /></ListItemIcon>
            <ListItemText>Orders</ListItemText>
          </ListItem>
        </Link>
        <Link to = '/customers' className={classes.link}>
          <ListItem button className="icon">
            <ListItemIcon><FaceIcon /></ListItemIcon>
            <ListItemText>Customers</ListItemText>
          </ListItem>
        </Link>
      </List>
      <Divider />
      <List>
        <Link to = '/staffs' className={classes.link}>
          <ListItem button className="icon">
            <ListItemIcon><AccessibilityIcon /></ListItemIcon>
            <ListItemText>Staff</ListItemText>
          </ListItem>
        </Link>
      </List>
    </React.Fragment>
  )
}

CategoryList.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired
  }).isRequired).isRequired,
  setCategoryId: PropTypes.func,
  handleDrawerClose: PropTypes.func,
}

export default CategoryList
