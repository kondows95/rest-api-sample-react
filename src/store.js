import rootReducer from './modules/index';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
export default createStore(rootReducer, applyMiddleware(thunk));