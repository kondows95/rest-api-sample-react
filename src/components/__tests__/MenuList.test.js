import React from 'react'
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MenuList from '../MenuList';
import { Provider } from 'react-redux'
import store from '../../store';
let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})
afterEach(() => {
  document.body.outerHTML = '<body></body>'
  container = null
})
describe("MenuList component", () => {
  it('matches the snapshot', () => {
    const MenuListSnapshot = renderer.create(
      <Provider store={store}>
        <Router>
          <MenuList />
        </Router>
      </Provider>
    ).toJSON();
    expect(MenuListSnapshot).toMatchSnapshot();
  });
});


describe("testing menu list", () => {
  it('testing menulist link', () => {
    act(() => {
      ReactDOM.render((
        <Provider store={store}>
          <Router>
            <MenuList />
          </Router>
        </Provider>
      ), container);
    });
    const linkArr = document.querySelectorAll('a');
    expect(linkArr[0].textContent.toUpperCase()).toBe("CATEGORIES")
    expect(linkArr[0].getAttribute("href")).toBe("/categories")
    expect(linkArr[1].textContent.toUpperCase()).toBe("ITEMS")
    expect(linkArr[1].getAttribute("href")).toBe("/")
    expect(linkArr[2].textContent.toUpperCase()).toBe("STAFF")
    expect(linkArr[2].getAttribute("href")).toBe("/staffs")
  });
 
})

