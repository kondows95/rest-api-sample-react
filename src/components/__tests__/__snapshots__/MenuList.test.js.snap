// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuList component matches the snapshot 1`] = `
Array [
  <ul
    className="MuiList-root MuiList-padding"
  >
    <a
      className="makeStyles-link-1"
      href="/categories"
      onClick={[Function]}
    >
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root icon MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 14c-1.66 0-3 1.34-3 3 0 1.31-1.16 2-2 2 .92 1.22 2.49 2 4 2 2.21 0 4-1.79 4-4 0-1.66-1.34-3-3-3zm13.71-9.37l-1.34-1.34a.9959.9959 0 0 0-1.41 0L9 12.25 11.75 15l8.96-8.96c.39-.39.39-1.02 0-1.41z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            <span>
              Categories
            </span>
          </span>
        </div>
      </div>
    </a>
    <a
      className="makeStyles-link-1"
      href="/"
      onClick={[Function]}
    >
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root icon MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 19V5c0-1.1-.9-2-2-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2zM8.5 13.5l2.5 3.01L14.5 12l4.5 6H5l3.5-4.5z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            <span>
              Items
            </span>
          </span>
        </div>
      </div>
    </a>
  </ul>,
  <hr
    className="MuiDivider-root"
  />,
  <ul
    className="MuiList-root MuiList-padding"
  >
    <a
      className="makeStyles-link-1"
      href="/staffs"
      onClick={[Function]}
    >
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root icon MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2zm9 7h-6v13h-2v-6h-2v6H9V9H3V7h18v2z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            <span>
              Staff
            </span>
          </span>
        </div>
      </div>
    </a>
  </ul>,
  <ul
    className="MuiList-root MuiList-padding"
  >
    <a
      className="makeStyles-link-1"
      href="/orderlist"
      onClick={[Function]}
    >
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root icon MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 15h18v-2H3v2zm0 4h18v-2H3v2zm0-8h18V9H3v2zm0-6v2h18V5H3z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            <span>
              Orders
            </span>
          </span>
        </div>
      </div>
    </a>
  </ul>,
]
`;
